lucet-spectest/src/lib.rs

```diff
@@ -226,49 +231,54 @@ fn step(script: &mut ScriptEnv, cmd: &CommandKind) -> Result<(), SpecTestError>
                 match res {
                     Err(ScriptError::RuntimeError(_luceterror)) => Ok(()),
                     Err(e) => Err(unexpected_failure(e)),
-                    Ok(_) => Err(SpecTestErrorKind::UnexpectedSuccess)?,
+                    Ok(_) => Err(Error::UnexpectedSuccess)?,
                 }
             }
-            _ => Err(SpecTestErrorKind::UnsupportedCommand)?,
+            _ => {
+                let message = format!("invoke {:?}", action);
+                Err(Error::UnsupportedCommand(message))?
+            }
         },
     }
 }
 
-fn check_retval(expected: &[Value], got: UntypedRetVal) -> Result<(), SpecTestError> {
+fn check_retval(expected: &[Value], got: UntypedRetVal) -> Result<(), Error> {
     match expected.len() {
         0 => {}
         1 => match expected[0] {
             Value::I32(expected) => {
                 if expected != got.as_i32() {
-                    Err(format_err!("expected {}, got {}", expected, got.as_i32()))
-                        .context(SpecTestErrorKind::IncorrectResult)?
+                    let message = format!("expected {}, got {}", expected, got.as_i32());
+                    Err(Error::IncorrectResult(message))?
                 }
             }
             Value::I64(expected) => {
                 if expected != got.as_i64() {
-                    Err(format_err!("expected {}, got {}", expected, got.as_i64()))
-                        .context(SpecTestErrorKind::IncorrectResult)?
+                    let message = format!("expected {}, got {}", expected, got.as_i64());
+                    Err(Error::IncorrectResult(message))?
                 }
             }
             Value::F32(expected) => {
                 if expected != got.as_f32() && !expected.is_nan() && !got.as_f32().is_nan() {
-                    Err(format_err!("expected {}, got {}", expected, got.as_f32()))
-                        .context(SpecTestErrorKind::IncorrectResult)?
+                    let message = format!("expected {}, got {}", expected, got.as_f32());
+                    Err(Error::IncorrectResult(message))?
                 }
             }
             Value::F64(expected) => {
                 if expected != got.as_f64() && !expected.is_nan() && !got.as_f64().is_nan() {
-                    Err(format_err!("expected {}, got {}", expected, got.as_f64()))
-                        .context(SpecTestErrorKind::IncorrectResult)?
+                    let message = format!("expected {}, got {}", expected, got.as_f64());
+                    Err(Error::IncorrectResult(message))?
                 }
             }
             Value::V128(_) => {
-                Err(format_err!("got unsupported SIMD V128 value"))
-                    .context(SpecTestErrorKind::UnsupportedCommand)?;
+                let message = format!("got unsupported SIMD V128 value");
```

Comment:
I love how you've outlined the `message` values in this file :heart_decoration: :woman_technologist:

Minor nit: We create a `String` using `format!`, even though the literal doesn't need to interpolate a value with `{}`. We can use `String::from` instead, [like this](https://doc.rust-lang.org/std/string/struct.String.html#examples-97)

```suggestion
                let message = String::from("got unsupported SIMD V128 value");
```

Reply:
Thanks.  I agonized a bit over this.  Was it better to consistently use `format!` for error messages, or use `String::new()` when interpolation isn't needed?  It seems I chose incorrectly.  I will replace all the places where I made this choice.

Reply:
For what it's worth, this is _totally_ a small style nit. I think incorrect is a strong word :slightly_smiling_face: